File: 1700. Number of Students Unable to Eat Lunch.js
// 1700. Number of Students Unable to Eat Lunch
//students = [1,1,0,0]; sandwiches = [0,1,0,1]
students = [1,1,1,0,0,1]; sandwiches = [1,0,0,0,1,1]
/**
 * @param {number[]} students
 * @param {number[]} sandwiches
 * @return {number}
 */
var countStudents = function(students, sandwiches) {
  while (students.length > 0) {
      if (students[0] === sandwiches[0]) {
          students.shift();
          sandwiches.shift();
      } else {
          students.push(students.shift());
      }

      if (!students.includes(sandwiches[0])) break;
  }
  return students.length;
};
console.log(countStudents(students, sandwiches), 3);

File: 1700. Number of Students Unable to Eat Lunch.rb
#* 1700. Number of Students Unable to Eat Lunch
#students = [1,1,0,0]; sandwiches = [0,1,0,1]
students = [1,1,1,0,0,1]; sandwiches = [1,0,0,0,1,1]
# @param {Integer[]} students
# @param {Integer[]} sandwiches
# @return {Integer}
def count_students(students, sandwiches)
    until students.empty?
      if students.first == sandwiches.first
        students.shift
        sandwiches.shift
      else
        students.push(students.shift)
      end
      break if students.all? {|s| s != sandwiches.first}

    end
    students.length
end
p count_students(students, sandwiches)